src/mingwX64Main/kotlin/File.kt

```diff
@@ -0,0 +1,316 @@
+package me.archinamon.fileio
+
+import kotlinx.cinterop.*
+import platform.windows.*
+
+// Difference between January 1, 1601 and January 1, 1970 in millis
+private const val EPOCH_DIFF = 11644473600000
+
+
+actual class File actual constructor(
+    private val pathname: String
+) {
+    private val fileSeparator
+        get() = if (Platform.osFamily == OsFamily.WINDOWS) "\\" else "/"
+
+    actual fun getParent(): String? {
+        return if (exists()) getAbsolutePath().substringBeforeLast(fileSeparator) else null
+    }
+
+    actual fun getParentFile(): File? {
+        return getParent()?.run(::File)
+    }
+
+    actual fun getName(): String {
+        return if (fileSeparator in pathname) {
+            pathname.split(fileSeparator).last(String::isNotBlank)
+        } else {
+            pathname
+        }
+    }
+
+    actual fun lastModified(): Long {
+        val handle = CreateFileA(
+            pathname,
+            GENERIC_READ,
+            0,
+            null,
+            OPEN_EXISTING,
+            FILE_ATTRIBUTE_NORMAL,
+            null
+        )
+        if (handle == INVALID_HANDLE_VALUE) {
+            return 0L
+        }
+
+        return try {
+            memScoped {
+                val ft = alloc<_FILETIME>()
+                val result = GetFileTime(handle, null, null, ft.ptr)
+                if (result == TRUE) {
+                    val st = alloc<_SYSTEMTIME>()
+                    val convertResult = FileTimeToSystemTime(ft.ptr, st.ptr)
+                    if (convertResult == TRUE) {
+                        val time = (ft.dwHighDateTime.toLong() shl 32) or ft.dwLowDateTime.toLong()
+                        (time / 10000) - EPOCH_DIFF
+                    } else 0L
+                } else 0L
+            }
+        } finally {
+            CloseHandle(handle)
+        }
+    }
+
+    actual fun mkdirs(): Boolean {
+        if (exists()) return false
+
+        if (getParentFile()?.exists() == false) {
+            getParentFile()?.mkdirs()
+        }
+
+        return SHCreateDirectoryExA(null, getAbsolutePath(), null) == ERROR_SUCCESS
```

Comment:
does such calls safe to call out of memScope?

Reply:
Yes because it does not alloc any memory for the parameters, it is safe.